include/ck_tile/host/reference/reference_pool.hpp

```diff
@@ -145,4 +146,46 @@ CK_TILE_HOST void reference_pool3d(const HostTensor<InDataType>& input,
     make_ParallelTensorFunctor(f, N, Do, Ho, Wo, C)(std::thread::hardware_concurrency());
 }
 
+/// @brief Validates that pooling indices correctly point to their corresponding input values
+/// @param input Input tensor
+/// @param output Output tensor (for value comparison)
+/// @param output_index Output indices tensor (to validate)
+/// @return true if all indices are valid, false otherwise
+template <typename InDataType, typename OutDataType, typename IndexDataType>
+CK_TILE_HOST bool validate_pool_indices(const HostTensor<InDataType>& input,
+                                        const HostTensor<OutDataType>& output,
+                                        const HostTensor<IndexDataType>& output_index)
+{
+    int mismatches = 0;
+
+    const auto input_size  = input.get_element_space_size();
+    const auto output_size = output.get_element_space_size();
+
+    // Validate each output position's index
+    for(int out_pos = 0; out_pos < static_cast<int>(output_size); out_pos++)
+    {
+        auto kernel_idx = output_index.mData[out_pos];
```

Comment:
I think that HostTensor should have overloaded array access operator. 

```suggestion
        auto kernel_idx = output_index[out_pos];
```